[Masonry] Improve the styling on the demos

hbjORbj opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui-org/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

The style of the div items used for demo of `Masonry` can be improved to be consistent with those used for demo of `Grid`, which can be found here: https://next.material-ui.com/components/grid/#main-content

**Before**:
https://deploy-preview-27439--material-ui.netlify.app/components/masonry/
![Screenshot 2021-08-25 at 13 55 43](https://user-images.githubusercontent.com/32841130/130797306-73f997a1-5c7d-402b-b633-11c20cf4f9e3.png)


**After**:
https://deploy-preview-27957--material-ui.netlify.app/components/masonry/
![Screenshot 2021-08-25 at 14 15 24](https://user-images.githubusercontent.com/32841130/130797292-374f6356-f448-4324-ae48-ca9286e675e1.png)

mnajdova commented:
I would propose maybe using bigger font for the numbers and putting it in the center of the box, but I would leave it to @danilo-leal :)

hbjORbj commented:
> I'm in for this restyling :) It looks much better indeed and it is consistent with other components.
> 
> > I would propose maybe using bigger font for the numbers and putting it in the center of the box
> 
> Regarding this @mnajdova's suggestion, I feel like the numbers font-size is alright but putting them in the center would probably be better indeed!

Thanks, I addressed your comment!

oliviertassinari reviewed (commented):
- The isolation of the style of the items to a `Item` component seems to be a clear win. It makes the code preview simpler to follow
- The older style made me think about https://strml.github.io/react-grid-layout/examples/0-showcase.html
- +1 for consistency between the different demos.
- The dark theme on https://next.material-ui.com/components/grid/ doesn't look awesome in my opinion. I struggle with the color contrast.

<img width="889" alt="Screenshot 2021-08-30 at 11 41 58" src="https://user-images.githubusercontent.com/3165635/131320647-264c3b90-d231-44da-93cc-7079e8078259.png">

- Regarding the new look and feel. I'm personally not a fan of the elevation, Google seems to be moving increasingly toward outlined (on its new product). But it looks great, it's its sill  heavily by Google on dashboard: firebase, google cloud console, google search console, google ads, so 👍

danilo-leal commented:
> The dark theme on https://next.material-ui.com/components/grid/ doesn't look awesome in my opinion. I struggle with the color contrast.

How about we tackle this point here outside of this PR? Since you're bringing up the Grid, there are probably other component demos, similar to the Grid and Masonry, we can make better but by doing so now, we'd be enlarging the scope here. It's very worth digging into it though, just looking to close this as its main objective has been reached.

@mnajdova @hbjORbj what do you think?!

danilo-leal reviewed (approved):
I believe that for now, we're good to go! About one of @oliviertassinari comments, this one:

> The dark theme on https://next.material-ui.com/components/grid/ doesn't look awesome in my opinion. I struggle with the color contrast.

Although it's technically in the scope of "improving Masonry styling" he brings up the Grid as another example of a color contrast issue. Since there are these two and probably other components with this problem, I think we could probably tackle them all in a separate PR?! We could then close this one for now, not enlarging its scope (by fixing the Grid and other components here). And then open a new one for addressing the issue, as it is very worth to do so. 

@mnajdova @hbjORbj what do you think?!

hbjORbj commented:
@danilo-leal I agree. I am merging it now!
